storage/Filesystems/FilesystemImpl.h

```diff
@@ -252,6 +253,32 @@ namespace storage
 
     }
 
+
+    class EnsureMounted : boost::noncopyable
+    {
+
+    public:
+
+	/**
+	 * Ensures that the filesystem is mounted somewhere.
+	 *
+	 * The mode is not enforced.
```

Comment:
Then wouldn't it be better to drop the Mode arguments and document that we cannot guarantee the mode anyway? That's certainly better for the current use case.